source/indigo_to_melodic.rst

```diff
@@ -0,0 +1,186 @@
+ROS Melodic + Ubuntu 18.04
+==========================
+
+Upgrading Your Robot to ROS Melodic + Ubuntu 18.04
+--------------------------------------------------
+.. WARNING::
+   Read this document in full to ensure you understand the procedures.  It is
+   not straightforward to go back to ROS Indigo/Ubuntu 14.04 after doing this.
+   Ensure your colleagues are on board with doing this upgrade.
+
+This document is a procedure for replacing the contents of your robot's SSD
+with an Ubuntu 18.04 install and ROS Melodic.
+
+Before Upgrade
+++++++++++++++
+
+Back up files from the robot!  There are a few categories of files to back up:
+
+#. Calibration and other robot-specific files. By convention, these are
+   all in ``/etc/ros/[indigo|melodic]/``
+#. Files relating to your research work
+#. Network hardware configuration (for troubleshooting)
+
+Below, we assume that after logging into the robot (e.g. via `ssh`) you back up
+files to a machine named HOST with username USER.
+
+For (1), we recommend doing::
+
+  tar -zcf fetch_robot_files.tar.gz /etc/ros/indigo/
+  scp fetch_robot_files.tar.gz USER@HOST:~/
+
+For (2), this may include workspaces, logs, and training data.
+
+For (3), this file may be useful::
+
+  scp /etc/udev/rules.d/70-persistent-net.rules USER@HOST:~/$(hostname)_udev_net_rules
+
+
+18.04 Install and Installing ROS/Fetch Packages
++++++++++++++++++++++++++++++++++++++++++++++++
+
+.. IMPORTANT::
+   Back up your files as described in the previous section
+
+#. **Runstop the robot**, to avoid unexpected movement of the robot.
+#. Download the latest 18.04 Ubuntu installer from http://releases.ubuntu.com/18.04/
+   (in these instructions we use the Desktop image, version 18.04.1). Install Ubuntu
```

Comment:
The actual TODO is to make the installer and negate the need for these steps :)  But I might take your suggestion here, and have manual steps for a server-based ubuntu setup instead of supporting an additional installer for that. (though it would potentially be super minimally different)